### PR TITLE
chaos(engine): use kind to setup k8s cluster in chaos test

### LINKED ISSUE #5738: coredns pod is pending in ci test
### What did you do?

test `Dataflow Engine Chaos` failed because of k3s setup failure. ref: https://github.com/pingcap/tiflow/runs/6718710616?check_suite_focus=true

### What did you expect to see?

_No response_

### What did you see instead?

Found we can setup k3s successfully if we use flannel as default backend, but chaos-mesh doesn't work normally with it.

```bash
Run kubectl rollout status --watch --timeout 600s deployment/coredns -n kube-system
  kubectl rollout status --watch --timeout 600s deployment/coredns -n kube-system
  shell: /bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    KUBECONFIG: /etc/rancher/k[3](https://github.com/pingcap/tiflow/runs/6718710713?check_suite_focus=true#step:6:3)s/k3s.yaml
Waiting for deployment spec update to be observed...
Waiting for deployment spec update to be observed...
Waiting for deployment "coredns" rollout to finish: 0 out of 1 new replicas have been updated...
Waiting for deployment "coredns" rollout to finish: 0 of 1 updated replicas are available...
error: timed out waiting for the condition
Error: Process completed with exit code 1.
```

### Versions of the cluster

Dataflow Engine version (run `df-master -V` or `df-executor -V`):

master@https://github.com/pingcap/tiflow/commit/fde7da36d370cfb675ce1a7f999ff1d7dd684257

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve?

Issue Number: close #5738

### What is changed and how it works?

Use [kind](https://kind.sigs.k8s.io/) to setup K8s cluster in chaos test, which is easier and more widely used than [k3s](https://k3s.io/)

#### Tests 

 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by lonng — APPROVED** [review]

File: .github/workflows/dataflow_engine_chaos.yaml
Comment (lance6716) [inline .github/workflows/dataflow_engine_chaos.yaml:6]: it's to test the effect of this PR, or we are really going to let every PR trigger chaos test?

**Comment by amyangfei** [comment]
/hold
remove the trigger of every pull request to master before merging this PR.

File: .github/workflows/dataflow_engine_chaos.yaml
Comment (amyangfei) [reply .github/workflows/dataflow_engine_chaos.yaml:6]: It is for test only.

**Review by lance6716 — APPROVED** [review]

File: .github/workflows/dataflow_engine_chaos.yaml
Comment (amyangfei) [inline .github/workflows/dataflow_engine_chaos.yaml:7]:
```suggestion
```

**Pull request merged**
